src/common/Logging/Extensions.cs

```diff
@@ -118,7 +118,7 @@ public static string Serialize(this object input)
             {
                 return JsonHelpers.SerializeObject(input, serializerSettings);
             }
-            catch (Exception)
+            catch (JsonException)

```

Comment:
From library log file I see this string "Serialization Error" message [bridge-library-2023-02-07-22-41-06-19390.txt](https://github.com/Azure/Bridge-To-Kubernetes/files/10680038/bridge-library-2023-02-07-22-41-06-19390.txt)

Stack trace below too:
---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Threading.Tasks.Task.GetExceptions(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCoreAsObject(Utf8JsonWriter writer, Object value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteUsingSerializer[TValue](Utf8JsonWriter writer, TValue& value, JsonTypeInfo jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Serialize[TValue](Utf8JsonWriter writer, TValue value, JsonSerializerOptions options)
   at Microsoft.BridgeToKubernetes.Common.Logging.Extensions.ExceptionConverter`1.Write(Utf8JsonWriter writer, TExceptionType value, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCoreAsObject(Utf8JsonWriter writer, Object value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteUsingSerializer[TValue](Utf8JsonWriter writer, TValue& value, JsonTypeInfo jsonTypeInfo)
   at System.Text.Json.JsonSerializer.WriteStringUsingSerializer[TValue](TValue& value, JsonTypeInfo jsonTypeInfo)
   at Microsoft.BridgeToKubernetes.Common.Logging.Extensions.Serialize(Object input)
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at Microsoft.BridgeToKubernetes.Common.Logging.LoggerBase.FormatMessage(String format, Object[] args)
   at Microsoft.BridgeToKubernetes.Common.Logging.LoggerBase.ExceptionInner(Exception e, EventLevel eventLevel, Boolean handled)
   at Microsoft.BridgeToKubernetes.Common.Logging.LoggerBase.Exception(Exception e, Boolean handled)
   at Microsoft.BridgeToKubernetes.Common.Logging.Log.<>c__DisplayClass19_0.<Exception>b__0(ILogger x)
   at System.Collections.Generic.IEnumerableExtensions.ExecuteForEach[T](IEnumerable`1 items, Action`1 action, CancellationToken cancellationToken)
   at Microsoft.BridgeToKubernetes.Common.Logging.Log._Call(Action`1 action)
   at Microsoft.BridgeToKubernetes.Common.Logging.Log.Exception(Exception e, Boolean handled)
   at Microsoft.BridgeToKubernetes.Common.DevHostAgent.DevHostAgentExecutorClient.<>c__DisplayClass14_1.<<ReversePortForwardStartAsync>b__0>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Threading.Tasks.Task.TrySetCanceled(CancellationToken tokenToRecord, Object cancellationException)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception exception, Task`1& taskField)
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.InvokeCoreAsync(String methodName, Type returnType, Object[] args, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(Action action, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Threading.Tasks.Task.TrySetCanceled(CancellationToken tokenToRecord, Object cancellationException)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetException(Exception exception, Task`1& taskField)
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.InvokeCoreAsyncCore(String methodName, Type returnType, Object[] args, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
--- End of stack trace from previous location ---

Reply:
I think because it is throwing TaskCenceledException after this change the catch is no longer catching the exception. I think should go back to generic exception. Although even after that we still need to see why it is trowing to begin with.

Reply:
I think we can use ExceptionConverters as workaround to solve this issue. Thoughts? we can add a UT to test this if it works or not. Thanks.
```
            try
            {
                serializerSettings.Converters.Add(new ExceptionConverter<Exception>());
                return JsonHelpers.SerializeObject(input, serializerSettings);
            }
            catch (Exception exception)
            {
                //return "Serialization Error";
                return JsonSerializer.Serialize(exception, serializerSettings);
            }

```

Reply:
Yeah that was my first approach. But I ran into the issue with custom converters being dependent on the same custom converter. The problem occurs when I tried to use `JsonSerializer.Serialize(exception, serializerSettings);` inside the Write function and I'm reusing the serializerSettings parameter from the method. It's a common problem in STJ.
But since Exceptions can have InnerExceptions that can have InnerExceptions and so on, the serializerSettings inside the Write function needs to know how to serializes Exception, but maby there is a way around this?

Reply:
@Eneuman I think it is because inside the custom coverter you are already serializing the message as JSON and again in the catch block you are doing serialize again might be the issue. May be try using writer.writeString(value.message) and see if it works ?

<img width="380" alt="image" src="https://user-images.githubusercontent.com/105889062/218670410-19be358c-6c1f-4d83-9016-cc6360de0a0e.png">

Reply:
@Eneuman I tried commenting out the serialize line, it works partially, I will keep looking into and post my findingss soon here.